### PR TITLE
Fix/linkta 321 remove firebase oauth flow from the related files

### PR DESCRIPTION
**Pull request by zhiyi15**
## Description
Removed Oauth flow from firebase.ts and PrelaunchForm.tsx

Fixes # (issue)321

## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] Other 🗒️ **\*\***\_\_**\*\***

## How Has This Been Tested?

Ran `npm run dev` to make sure everything else is working normally

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: app/config/firebase.ts
Comment (kinskiwu) [inline app/config/firebase.ts]: Since the db functionality now lives in the signup form I don't think you need to keep this import anymore.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (kinskiwu) [inline app/components/main-content/PrelaunchSignUpForm.tsx]: Since Darren is working on styling it may be better not to make any changes to the UI unless it is related to this PR.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (kinskiwu) [inline app/components/main-content/PrelaunchSignUpForm.tsx]: Updates related to Firestore are implemented in #39, unless it is causing errors it may be better to leave it to the other ticket and not make changes in this PR.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (kinskiwu) [inline app/components/main-content/PrelaunchSignUpForm.tsx:5]: Are we using Dispatch and SetStateAction in this component?

**Review by kinskiwu — CHANGES_REQUESTED** [review]
Thanks for the good work! I left some comments for small changes.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (zhiyi15) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: I actually made these changes before knowing about #39 PR. I thought it would be fine to leave it since this PR would probably be merge first before merging #39. But i will change it back to untouched.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (zhiyi15) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: Certainly, i will change it back.

File: app/config/firebase.ts
Comment (zhiyi15) [reply app/config/firebase.ts]: getFirestore should stay in this file because it is the entry point for all firebase service, but i will remove the rest of the imports.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (zhiyi15) [reply app/components/main-content/PrelaunchSignUpForm.tsx:5]: I'm not sure about Dispatch and SetStateAction as it wasn't related to Oauth flow. 
But i can see that Dispatch and SetStateAction are used here
`interface PrelaunchSignUpFormProps {
  setFlowState: Dispatch<SetStateAction<FlowState>>;
}`.

Comment (kinskiwu) [reply app/components/main-content/PrelaunchSignUpForm.tsx:5]: I see. Maybe better keep it there for now.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (kinskiwu) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: Thanks!

**Review by kinskiwu — APPROVED** [review]

**Comment by dcpavel** [comment]
Have we tested that signing up via email still works after these changes are made?

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (ejwoods) [inline app/components/main-content/PrelaunchSignUpForm.tsx]: I think this section and heading component can be removed now that we only have one sign up section.

**Review by ejwoods — CHANGES_REQUESTED** [review]
Looks good! I left one comment but ready to approve after that's resolved.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (zhiyi15) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: That’s weird, I thought I removed the buttons for Oauth already. Will definitely remove it

Comment (zhiyi15) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: Wait, I did remove the oauth section, is that what you meant?

**Comment by zhiyi15** [comment]
> Have we tested that signing up via email still works after these changes are made?

Yes I have tested. Due to the current firestore security rule is set for only allow authenticated user to read and write into DB, so technically signing up via email is not working yet.

File: app/components/main-content/PrelaunchSignUpForm.tsx
Comment (ejwoods) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: You removed the section with the OAuth buttons, but we can also remove the `<section>` and `<h3>` HTML elements for the "sign up with email" section. These elements were only needed to separate the OAuth and Email sections from one another - now that there's only one section, we don't need the elements distinguishing them.

Comment (ejwoods) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: Ah, I see this was mentioned in a previous comment - https://github.com/Linkta-org/landing-page/pull/38#discussion_r1573476276

I would consider fixing this part of the UI related to this PR. But if this UI fix is not implemented here, we should make a ticket to ensure it gets done @kinskiwu

Comment (kinskiwu) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: @dcpavel Would removing this section have a big impact on your styling? I think it is ok to remove the h3 but we may want to keep the section tag for styling.

Comment (kinskiwu) [reply app/components/main-content/PrelaunchSignUpForm.tsx]: @zhiyi15 I looked at Darren's ticket and it should be safe to remove h3: or sign up with email. Pls keep the section tag.

**Review by ejwoods — APPROVED** [review]
Removed the `<h3>` element I previously requested changes on but left the `<section>` element so as not to conflict with PR #23.

**Pull request merged**
